applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -752,6 +753,40 @@ static void audio_datapath_just_in_time_check_and_adjust(uint32_t sdu_ref_us)
 	}
 }
 
+/**
+ * @brief Update sdu_ref_us so that drift compensation can work correctly.
+ *
+ * @note This function is only valid for gateway using I2S as audio source
+ *       and unidirectional audio stream (gateway to headset(s)).
+ *
+ * @param sdu_ref_us    ISO timestamp reference from BLE controller.
+ * @param adjust        Indicate if the sdu_ref should be used to adjust timing.
+ */
+static void audio_datapath_sdu_ref_update(const struct zbus_channel *chan)
+{
+	if (IS_ENABLED(CONFIG_AUDIO_SOURCE_I2S)) {
+		uint32_t sdu_ref_us;
+		bool adjust;
+		const struct sdu_ref_msg *msg;
+
+		msg = zbus_chan_const_msg(chan);
+		sdu_ref_us = msg->timestamp;
+		adjust = msg->adjust;
+
+		if (ctrl_blk.stream_started) {
+			ctrl_blk.previous_sdu_ref_us = sdu_ref_us;
+
+			if (adjust) {
+				audio_datapath_just_in_time_check_and_adjust(sdu_ref_us);
+			}
+		} else {
+			LOG_WRN("Stream not startet - Can not update sdu_ref_us");
+		}
+	}
+}
+
+ZBUS_LISTENER_DEFINE(sdu_ref_msg_listen, audio_datapath_sdu_ref_update);
```

Comment:
Would it be better to  check if we are using CONFIG_AUDIO_SOURCE_I2S then we enable ZBUS_LISTENER_DEFINE?

Reply:
I take the thumb back :P If we use ```#if CONFIG_AUDIO_SOURCE_I2S``` around the listener_define we would get a warning in the compiler when it is not set, since the function is defined but not used.

Reply:
We have removed almost all #if from audio_datapath, so I suggest not adding #if. Use another approach, if needed.
Maybe we should add a CONFIG_AUDIO_SOURCE_I2S guard in broadcast_source_send()?

Reply:
It shouldn't need to know who will process it. I think the current solution is good